### PR TITLE
Upgrading versions and minor change to xreg.

### PR DESCRIPTION
**Pull request by rajatsen91**

### CODE REVIEW

File: src/timesfm/xreg_lib.py [hunk 1]
```diff
@@ -372,25 +341,22 @@ def create_covariate_matrix(
...
+      x_std = np.where((w := np.std(x_train, axis=0, keepdims=True)) > _TOL, w,
+                       1.0)
       x_train = [(x_train - x_mean) / x_std]
       x_test = [(x_test - x_mean) / x_std]
 
     # Categorical features. Encode one by one.
     one_hot_encoder = preprocessing.OneHotEncoder(
         drop=one_hot_encoder_drop,
-        sparse=False,
+        sparse_output=False,
```
Comment (siriuz42) [inline src/timesfm/xreg_lib.py:352]: Thanks for revising this.

File: pyproject.toml [hunk 2]
```diff
@@ -30,15 +30,15 @@ include = [
...
-scikit-learn = "1.0.2"
+einshape = ">=1.0.0"
+numpy = ">=1.26.4"
+pandas = ">=2.1.4"
+paxml = ">=1.4.0"
+utilsforecast = ">=0.1.10"
+jax = {version = ">=0.4.26", extras = ["cuda12"]}
+jaxlib = ">=0.4.26"
+huggingface_hub = {version = ">=0.23.0", extras = ["cli"]}
+scikit-learn = ">=1.0.2"
```
Comment (siriuz42) [inline pyproject.toml:41]: scikit-learn = ">=1.2.2"

1.2.2 is the first version that introduced sparse_output as per the [API doc](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html).

**Review by siriuz42 — CHANGES_REQUESTED** [review]
Thanks for taking care of this. PTAL.

File: src/timesfm/xreg_lib.py (hunk 1 above)
Comment (rajatsen91) [reply src/timesfm/xreg_lib.py:352]: done.

**Pull request merged**